core/controllers/practice_sessions.py

```diff
@@ -52,16 +54,22 @@ def get(self, topic_name):
         # Topic cannot be None as an exception will be thrown from its decorator
         # if so.
         topic = topic_fetchers.get_topic_by_name(topic_name)
+        subtopic_titles_list = self.request.get('selected_subtopics')
+        selected_subtopic_titles = json.loads(subtopic_titles_list)
```

Comment:
try/except here, return 400/404 if invalid format (remember, don't trust any input that comes in from the outside)

Reply:
Using split now to get IDs from string, in which cause even if it is not a correct string,it's fine as it would just be considered as an invalid subtopic ID and never matched.